sklearn/metrics/scorer.py

```diff
@@ -66,6 +67,34 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def _get_labels_from_estimator(self, estimator, target_type):
+        """Check if estimator has a classes argument and if scorer has a
+        labels argument and return new kwargs for scorer if label not already
+        set in make score initialization"""
+        est_has_classes = hasattr(estimator, "classes_")
+        scorer_has_labels = "labels" in signature(
+                                            self._score_func).parameters
+
+        if est_has_classes and scorer_has_labels:
+            classes = getattr(estimator, "classes_", None)
```

Comment:
Indeed, `classes_` it is. In fact, I would like to see this proposed as a separate enhancement to scorers, rather than dependent on us adopting the `classes` parameter in classifiers. I'd consider it a bug fix because the scoring function would now become dependent on the set of labels present in training, rather than those dependent in the test set.

WDYT?